pkg/issuer/venafi/client/venaficlient.go

```diff
@@ -116,14 +119,18 @@ func New(namespace string, secretsLister internalinformers.SecretLister, issuer
 
 	instrumentedVCertClient := newInstumentedConnector(vcertClient, metrics, logger)
 
-	return &Venafi{
+	v := &Venafi{
 		namespace:     namespace,
 		secretsLister: secretsLister,
 		vcertClient:   instrumentedVCertClient,
 		cloudClient:   cc,
 		tppClient:     tppc,
 		config:        cfg,
-	}, nil
+	}
+	if err := v.VerifyCredentials(); err != nil {
+		return nil, err
+	}
+	return v, nil
```

Comment:
This causes duplication of the API requests in the Venafi Issuer controller, because in `Setup` in  `pkg/issuers/venafi` we already call `VerifyCredentials`....as a way of checking the readiness of the Issuer resource.

Reply:
@wallrj as discussed in person, the reason this call is here is because we turned off the implicit authentication when calling `New` above. This call ensures we leave the function having actually authenticated to Venafi, as we did before this change.